File: Examples/ProfileExtrusion/ProfileExtrusion.cpp
#include <iostream>

#include <psDomain.hpp>
#include <psExtrudeProfile.hpp>
#include <psSmartPointer.hpp>

int main() {
  using NumericType = double;
  static constexpr int D = 3;

  const NumericType gridDelta = 0.05;
  const NumericType extrusionLength = 5.;

  auto geometry = psSmartPointer<psDomain<NumericType, D>>::New();

  std::vector<std::array<NumericType, 2>> profile;
  profile.emplace_back(std::array{1., 1.});
  profile.emplace_back(std::array{1., -1.});
  profile.emplace_back(std::array{-1., -1.});
  profile.emplace_back(std::array{-1., 1.});

  psExtrudeProfile<NumericType, D>(geometry, gridDelta, profile,
                                   extrusionLength, true)
      .apply();

  std::cout << geometry->getLevelSets()->size() << std::endl;
  geometry->printSurface("output.vtp");
}

File: Examples/ProfileExtrusion/ProfileExtrusion.py
import viennals3d as vls
import viennaps3d as vps


def main() -> None:
    grid_delta = 0.05
    extrusion_length = 5.0
    x_extent = 2.2
    y_extent = 2.2

    geometry = vps.psDomain()
    profile = [
        [1.0, 1.0],
        [1.0, -1.0],
        [-1.0, -1.0],
        [-1.0, 1.0],
    ]

    vps.psExtrudeProfile(geometry, grid_delta, x_extent, y_extent, profile,
                         extrusion_length).apply()
    geometry.printSurface("output.vtp")


if __name__ == "__main__":
    main()
